src/libstd/net/ip.rs

```diff
@@ -822,6 +822,14 @@ impl From<[u8; 16]> for Ipv6Addr {
     }
 }
 
+#[stable(feature = "ipv6_from_segments", since = "1.15.0")]
+impl From<[u16; 8]> for Ipv6Addr {
+    fn from(segments: [u16; 8]) -> Ipv6Addr {
+        let [a, b, c, d, e, f, g, h] = segments;
```

Comment:
I was under the impression that this is just providing aliases for each of the indexed variables, but I could be wrong. I assumed that bound variables and registers/stack slots was not a 1-1 mapping.